### PR TITLE
`cors` option was not working - #595

### PR DESCRIPTION
**Pull request by mbana**
See: <https://github.com/kubeshop/kusk-gateway/issues/595>.

Fixes kubeshop/kusk-gateway#595.

`examples/cors/*`
----------------------

Add `CORS` examples.

Signed-off-by: Mohamed Bana <mohamed@bana.io>

---

Example runs:

`examples/cors/cors-1`
----------------------

```
$ curl -v 192.168.49.2/uuid 
*   Trying 192.168.49.2:80...
* Connected to 192.168.49.2 (192.168.49.2) port 80 (#0)
> GET /uuid HTTP/1.1
> Host: 192.168.49.2
> User-Agent: curl/7.82.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< content-length: 47
< content-type: application/json; encoding=utf-8
< date: Tue, 04 Oct 2022 17:00:19 GMT
< x-envoy-upstream-service-time: 0
< access-control-allow-origin: *
< access-control-max-age: 86200
< access-control-allow-credentials: true
< server: envoy
< 
* Connection #0 to host 192.168.49.2 left intact
{"uuid":"6694d2c4-22ac-4208-a007-2939487f6999"}
```

`examples/cors/cors-2`
----------------------

```
$ curl -v 192.168.49.2/hello
*   Trying 192.168.49.2:80...
* Connected to 192.168.49.2 (192.168.49.2) port 80 (#0)
> GET /hello HTTP/1.1
> Host: 192.168.49.2
> User-Agent: curl/7.82.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< x-powered-by: Express
< content-type: application/json; charset=utf-8
< content-length: 48
< etag: W/"30-nU8AHUA5JUt8Fkoij7YIr9iKO7o"
< date: Tue, 04 Oct 2022 17:00:18 GMT
< x-envoy-upstream-service-time: 1
< access-control-allow-origin: https://example.org,https://httpbin.org
< access-control-max-age: 86200
< access-control-allow-credentials: true
< server: envoy
< 
* Connection #0 to host 192.168.49.2 left intact
{"message":"Hello from an implemented service!"}
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,6 +154,8 @@ docker-build: ## Build docker image with the manager.
 	eval $$(minikube docker-env --profile kgw); docker build \
 		--tag ${MANAGER_IMG} \
 		--tag kusk-gateway:latest \
+		--tag kubeshop/kusk-gateway:latest \
```
Comment (mbana) [inline Makefile:157]: Without this line I cannot get `make create-env` to pickup my local changes.

File: Makefile [hunk 2]
```diff
@@ -154,6 +154,8 @@ docker-build: ## Build docker image with the manager.
...
+		--tag kubeshop/kusk-gateway:$(shell git describe --tags $(shell git rev-list --tags --max-count=1)) \
```
Comment (mbana) [inline Makefile:158]: Without this line I cannot get `make create-env` to pickup my local changes.

See: <https://github.com/kubeshop/kusk-gateway/pull/746/files#r987136390>.

One of these lines is having an effect on my local changes being built or not.

File: internal/envoy/cors/cors.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		route.ResponseHeadersToAdd = append(route.ResponseHeadersToAdd, &envoy_config_core_v3.HeaderValueOption{
+			Header: &envoy_config_core_v3.HeaderValue{
+				Key:   HeaderAccessControlAllowCredentials,
+				Value: "true",
+			},
+			Append: wrapperspb.Bool(false),
+		})
+	}
+
+	fmt.Printf("ConfigureCORSOnRoute.%s: origins=%v\n", funcName, origins)
```
Comment (jasmingacic) [inline internal/envoy/cors/cors.go:91]: probably for future reference we should wrap this into logger

**Review by jasmingacic — APPROVED** [review]
Ship it

**Review by jasmingacic — DISMISSED** [review]

File: internal/envoy/cors/cors.go (hunk 3 above)
Comment (mbana) [reply internal/envoy/cors/cors.go]: I've changed it to use `logger` now.

**Review by jasmingacic — APPROVED** [review]

**Pull request merged**